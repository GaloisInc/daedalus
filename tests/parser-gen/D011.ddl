-- test Many with constant bound

def numBase (base : int) (ds : [ int ]) =
  for (val = 0; d in ds)
    (val * base + d)

def Numeral = Match1 ('0' .. '9')

def Digit = { @d = Numeral ; ^ d - '0' as int }

def UnsignedNum digs = {
  @ds = Many digs Digit ;
  ^ numBase 10 ds
}


def FL = Choose1 {
  unknown_length = Many 12 (Match1 '9');
  -- TODO put interval bounds here:
  file_length = UnsignedNum 12 ;
}


def Main =
  { b1 = FL
  ; b2 = UnsignedNum 6
  }
