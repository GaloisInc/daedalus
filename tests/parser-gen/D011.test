dump
D011.ddl
--gen
