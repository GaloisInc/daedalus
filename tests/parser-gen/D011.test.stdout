**********************************
***** Strict LLA Transitions *****
**********************************
Stack:
    *
    Main callsite __START__
    FL callsite Main (25,10)-(25,11)
    UnsignedNum callsite FL (20,17)-(20,30)
    Many
Start: UnsignedNum (12,9)-(12,23)
SlkCfg{ q:38, ctrl:[?, ?, ?, ?, *], sem:[?, ?, ], inp:--- }
DTrans [
  ( [(68),]
  , HeadInput ['0','9']
  , Resolution (NotAmbiguous)
  ),
]

Stack:
    *
    Main callsite __START__
    UnsignedNum callsite Main (26,10)-(26,22)
    Many
Start: UnsignedNum (12,9)-(12,23)
SlkCfg{ q:38, ctrl:[?, ?, ?, *], sem:[?, ?, ], inp:--- }
DTrans [
  ( [(24),]
  , AcceptingPath
  , NotAmbiguous
  ),
]

Stack:
    *
Start: Main (24,1)-(27,3)
SlkCfg{ q:1, ctrl:[*], sem:[*], inp:--- }
DTrans [
  ( [(51),]
  , HeadInput ['0','8']
  , Resolution (NotAmbiguous)
  ),
  ( [(51),(22),]
  , HeadInput ['9']
  , DTrans [
      ( [(31),]
      , HeadInput ['0','8']
      , Resolution (NotAmbiguous)
      ),
      ( [(31),(7),]
      , HeadInput ['9']
      , DTrans [
          ( [(31),]
          , HeadInput ['0','8']
          , Resolution (NotAmbiguous)
          ),
          ( [(31),(7),]
          , HeadInput ['9']
          , DTrans [
              ( [(31),]
              , HeadInput ['0','8']
              , Resolution (NotAmbiguous)
              ),
              ( [(31),(7),]
              , HeadInput ['9']
              , DTrans [
                  ( [(31),]
                  , HeadInput ['0','8']
                  , Resolution (NotAmbiguous)
                  ),
                  ( [(31),(7),]
                  , HeadInput ['9']
                  , DTrans [
                      ( [(31),]
                      , HeadInput ['0','8']
                      , Resolution (NotAmbiguous)
                      ),
                      ( [(31),(7),]
                      , HeadInput ['9']
                      , DTrans [
                          ( [(31),]
                          , HeadInput ['0','8']
                          , Resolution (NotAmbiguous)
                          ),
                          ( [(31),(7),]
                          , HeadInput ['9']
                          , DTrans [
                              ( [(31),]
                              , HeadInput ['0','8']
                              , Resolution (NotAmbiguous)
                              ),
                              ( [(31),(7),]
                              , HeadInput ['9']
                              , DTrans [
                                  ( [(31),]
                                  , HeadInput ['0','8']
                                  , Resolution (NotAmbiguous)
                                  ),
                                  ( [(31),(7),]
                                  , HeadInput ['9']
                                  , DTrans [
                                      ( [(31),]
                                      , HeadInput ['0','8']
                                      , Resolution (NotAmbiguous)
                                      ),
                                      ( [(31),(7),]
                                      , HeadInput ['9']
                                      , DTrans [
                                          ( [(31),]
                                          , HeadInput ['0','8']
                                          , Resolution (NotAmbiguous)
                                          ),
                                          ( [(31),(7),]
                                          , HeadInput ['9']
                                          , DTrans [
                                              ( [(31),]
                                              , HeadInput ['0','8']
                                              , Resolution (NotAmbiguous)
                                              ),
                                              ( [(31),(7),]
                                              , HeadInput ['9']
                                              , DTrans [
                                                  ( [(68),(55),]
                                                  , HeadInput ['0','9']
                                                  , Resolution (Ambiguous)
                                                  ),
                                                ]
                                              ),
                                            ]
                                          ),
                                        ]
                                      ),
                                    ]
                                  ),
                                ]
                              ),
                            ]
                          ),
                        ]
                      ),
                    ]
                  ),
                ]
              ),
            ]
          ),
        ]
      ),
    ]
  ),
]

**********************
**** Extended LLA ****
**********************
("DataDepTrans",2)
("Lookahead-0",3)
("Lookahead-1",54)
("Lookahead-ambiguous-13",3)
("abort-1",3)
("abort-7",1)

Total nb states: 66

**********************
***** Strict LLA *****
**********************
("Lookahead-0",1)
("Lookahead-1",17)
("Lookahead-ambiguous-13",1)

Total nb DFAs: 19


Warning: LL(*) failures:

Stack:
    *
Start: Main (24,1)-(27,3)
SlkCfg{ q:1, ctrl:[*], sem:[*], inp:--- }
DTrans [
  ( [(51),]
  , HeadInput ['0','8']
  , Resolution (NotAmbiguous)
  ),
  ( [(51),(22),]
  , HeadInput ['9']
  , DTrans [
      ( [(31),]
      , HeadInput ['0','8']
      , Resolution (NotAmbiguous)
      ),
      ( [(31),(7),]
      , HeadInput ['9']
      , DTrans [
          ( [(31),]
          , HeadInput ['0','8']
          , Resolution (NotAmbiguous)
          ),
          ( [(31),(7),]
          , HeadInput ['9']
          , DTrans [
              ( [(31),]
              , HeadInput ['0','8']
              , Resolution (NotAmbiguous)
              ),
              ( [(31),(7),]
              , HeadInput ['9']
              , DTrans [
                  ( [(31),]
                  , HeadInput ['0','8']
                  , Resolution (NotAmbiguous)
                  ),
                  ( [(31),(7),]
                  , HeadInput ['9']
                  , DTrans [
                      ( [(31),]
                      , HeadInput ['0','8']
                      , Resolution (NotAmbiguous)
                      ),
                      ( [(31),(7),]
                      , HeadInput ['9']
                      , DTrans [
                          ( [(31),]
                          , HeadInput ['0','8']
                          , Resolution (NotAmbiguous)
                          ),
                          ( [(31),(7),]
                          , HeadInput ['9']
                          , DTrans [
                              ( [(31),]
                              , HeadInput ['0','8']
                              , Resolution (NotAmbiguous)
                              ),
                              ( [(31),(7),]
                              , HeadInput ['9']
                              , DTrans [
                                  ( [(31),]
                                  , HeadInput ['0','8']
                                  , Resolution (NotAmbiguous)
                                  ),
                                  ( [(31),(7),]
                                  , HeadInput ['9']
                                  , DTrans [
                                      ( [(31),]
                                      , HeadInput ['0','8']
                                      , Resolution (NotAmbiguous)
                                      ),
                                      ( [(31),(7),]
                                      , HeadInput ['9']
                                      , DTrans [
                                          ( [(31),]
                                          , HeadInput ['0','8']
                                          , Resolution (NotAmbiguous)
                                          ),
                                          ( [(31),(7),]
                                          , HeadInput ['9']
                                          , DTrans [
                                              ( [(31),]
                                              , HeadInput ['0','8']
                                              , Resolution (NotAmbiguous)
                                              ),
                                              ( [(31),(7),]
                                              , HeadInput ['9']
                                              , DTrans [
                                                  ( [(68),(55),]
                                                  , HeadInput ['0','9']
                                                  , Resolution (Ambiguous)
                                                  ),
                                                ]
                                              ),
                                            ]
                                          ),
                                        ]
                                      ),
                                    ]
                                  ),
                                ]
                              ),
                            ]
                          ),
                        ]
                      ),
                    ]
                  ),
                ]
              ),
            ]
          ),
        ]
      ),
    ]
  ),
]

*******  Found Ambiguity  *******
  Start: Stack:
  Paths :
  -  
     *
     Main callsite __START__
     UnsignedNum callsite Main (26,10)-(26,22)
     Many
     Digit callsite UnsignedNum (12,19)-(12,23)
     Numeral callsite Digit (9,20)-(9,26)
     MATCH: Numeral (7,15)-(7,32)
  -  
     *
     Main callsite __START__
     UnsignedNum callsite Main (26,10)-(26,22)
     Many
     Digit callsite UnsignedNum (12,19)-(12,23)
     Numeral callsite Digit (9,20)-(9,26)
     MATCH: Numeral (7,15)-(7,32)
********  input witness  ********
9999999999990-9
